### PR TITLE
Bug 1948998: style fixes by GoLand and golangci-lint

### PR DESCRIPTION
**Pull request by Sergey1011010**
Run `golangci-lint run --skip-dirs-use-default --fix`, clicked a few auto-fix buttons in GoLand, and checked it manually.

## Categories

- [ ] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [X] Others - refactoring

## Unit Tests

Some unit tests were affected, it changed code structure, but not the logic. 

## References

https://issues.redhat.com/browse/???
https://bugzilla.redhat.com/show_bug.cgi?id=1948998
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/config/configobserver/configobserver_test.go [hunk 1]
```diff
@@ -28,10 +28,10 @@ func Test_ConfigObserver_ChangeSupportConfig(t *testing.T) {
 	}{
 		{name: "interval too short",
 			config: map[string]*corev1.Secret{
-				pullSecretKey: &corev1.Secret{Data: map[string][]byte{
+				pullSecretKey: {Data: map[string][]byte{
 					".dockerconfigjson": nil,
 				}},
-				supportKey: &corev1.Secret{Data: map[string][]byte{
+				supportKey: {Data: map[string][]byte{
```
Comment (tremes) [inline pkg/config/configobserver/configobserver_test.go:34]: Hmm I think I like the explicit definition more.

File: pkg/gather/clusterconfig/netnamespaces_test.go [hunk 2]
```diff
@@ -76,9 +76,7 @@ func createNetNamespaces(ctx context.Context, t *testing.T, n *networkfake.Clien
 	for _, ns := range namespaces {
 		_, err := n.NetworkV1().NetNamespaces().Create(ctx, ns, metav1.CreateOptions{})
 		if err != nil {
-			if err != nil {
-				t.Fatal("unable to create fake netnamespace", err)
-			}
+			t.Fatal("unable to create fake netnamespace", err)
```
Comment (tremes) [inline pkg/gather/clusterconfig/netnamespaces_test.go:79]: Lol...this was probably my work.

File: pkg/config/configobserver/configobserver_test.go (hunk 1 above)
Comment (rluders) [reply pkg/config/configobserver/configobserver_test.go:34]: I don't know. I think that, once the parent object is already a `corev1.Secret` by doing it two times, it is just overhead. So, IMHO, the change looks good.

**Review by rluders — APPROVED** [review]
/lgtm

**Pull request merged**